invisible_cities/types/ic_types_test.py

```diff
@@ -1,55 +1,69 @@
 import numpy as np
 
 from . ic_types          import minmax
+from . ic_types_c        import minmax as minmax_c
 from . ic_types          import xy
-from . ic_types_c        import xy as cxy
-from . ic_types_c        import minmax as cminmax
+from . ic_types_c        import xy     as xy_c
 
 from pytest import raises
+from pytest import mark
 
 from hypothesis            import given
 from hypothesis.strategies import floats
+from hypothesis.strategies import one_of
 from hypothesis.strategies import builds
 
 
-def make_minmax(a,b):
+def make_minmax(a, b):
     # Ensure that arguments respect the required order
     if a > b: a, b = b, a
     return minmax(a, b)
 
-def make_xy(a,b):
-    return xy(a,b)
 
-def make_cminmax(a,b):
+def make_minmax_c(a,b):
     # Ensure that arguments respect the required order
     if a > b: a, b = b, a
-    return cminmax(a, b)
+    return minmax_c(a, b)
+
+
+def make_xy(a,b):
```

Comment:
Why do we want `make_xy` and `make_xy_c` as synonyms for `xy` and `xy_c`?

Reply:
We don't. I didn't realize it was purely an alias.

Reply:
Well, strictly speaking, it's *not* an alias, because `make_xy` is a function, while `xy` is a type. Consequently, even though

    make_xy(a,b) == xy(a,b)

and

    type(make_xy(a,b)) == type(xy(a,b))

unfortunately

    type(make_xy(1,2)) != make_xy

whereas, quite sensibly

    type(xy(1,2)) == xy

Reply:
I am not very good with the nomenclature, but you get the point. The correct word would be... a wrapper?

Reply:
My point isn't about nomenclature. Yes, it's a wrapper, and it's (almost) a synonym, and (almost) an alias. But that's *not* the point.

The point is that it's utterly useless. It only wastes a couple of lines of code, and breaks some functionality.

I'm guessing that you were motivated to write it because you wanted to pass a callable to hypothesis' `builds`. The `xy` types themselves (like all types) are already perfectly good callables themselves (otherwise you wouldn't be able to call them inside your `make_xy` functions!). Like everything else in Python, they are first-class objects (just data), so you can pass them or store them or do whatever you want with them, without the need to wrap them.

In general, when you find yourself writing something of the form 

    def wrapper(*args, **kwds):
        return implementation(*args, **kwds)

or

    lambda *args, **kwds : implementation(*args, **kwds)

then **don't bother** and just use `implementation` itself, directly.

In both the above samples you are creating a function which takes some inputs, passes them on (without looking at them or modifying them at all) to `implementation` and then returns (without looking at or modifying it) whatever `implementation` returns. It matters not one bit whether `implementation` is a type or not.

That's a pointless waste of lines of code, CPU cycles, key presses, developer brain cycles and reader brain cycles.

Reply:
I agree 100%, but I didn't write that. I simply didn't realize that function was *just* a wrapper and used it as it was. I will remove it.

Reply:
You mean you didn't write the functions `make_xy` and `make_xy_c`?

Reply:
Exactly. I did rename the latter, though.

Reply:
Ah, OK, at first glance it looked like they appeared in your last commit, but looking more closely I see that they were there already before.

Looks like @jjgomezcadenas made them in mid-June.

@jjgomezcadenas Please do not write pointless wrappers around types.